File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/storage/ingrediente/IngredienteStorageServiceXML.java
package storage.ingrediente;

import config.ConfigApp;
import dto.ingrediente.ListaIngredienteDto;
import mapper.IngredienteMapper;
import models.Ingrediente;
import org.simpleframework.xml.core.Persister;
import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class IngredienteStorageServiceXML implements IngredienteStorageService{

    ConfigApp configApp = ConfigApp.getInstance();
    File file = new File(configApp.getAPP_DATA()+File.separator+"ingrediente.xml");

    public IngredienteStorageServiceXML() throws IOException {
    }

    Persister persister = new Persister();

    @Override
    public void saveAll(List<Ingrediente> entities) throws Exception {
        try {
            persister.write(IngredienteMapper.toDto(entities), file);
        }catch (Exception e){
            System.err.println(e.getMessage());
        }
    }

    @Override
    public List<Ingrediente> loadAll() throws Exception {
        List<Ingrediente> lista = new ArrayList<>();
        if(!file.exists()) return lista;
        try{
            lista = IngredienteMapper.toIngredientes(persister.read(ListaIngredienteDto.class, file));
        }catch (Exception e){
            System.err.println(e.getMessage());
        }
        return lista;
    }
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/storageService/Ingrediente/IngredienteStorageServiceXML.kt
package storageService.Ingrediente

import config.ConfigApp
import dto.ingrediente.ListaIngredientesDto
import mapper.toDto
import mapper.toIngrediente
import model.Ingrediente
import mu.KotlinLogging
import org.simpleframework.xml.core.Persister
import java.io.File

class IngredienteStorageServiceXML: IngredienteStorageService {

    private val config = ConfigApp
    private val file = File(config.APP_DATA+File.separator+"ingrediente.xml")
    private val logger = KotlinLogging.logger {  }

    private val serializer = Persister()

    override fun saveAll(entities: List<Ingrediente>) {
        logger.debug { "Se guardan todos los ingredientes en el archivo XML." }
        serializer.write(entities.toDto(), file)
    }

    override fun loadAll(): List<Ingrediente> {
        logger.debug { "Se cargan todos los ingredientes del archivo XML." }
        if(!file.exists()) return emptyList()
        return serializer.read(dto.ingrediente.ListaIngredientesDto::class.java, file).toIngrediente()
    }
}